src/hdmf/backends/hdf5/h5_utils.py

```diff
@@ -160,6 +160,11 @@ def __init__(self, **kwargs):
 
     def __read(self, path):
         s = self.__group[path][()]
+
```

Comment:
@t-b have you run into issues with this?

Reply:
A better fix might be to make H5SpecWriter.__write always write JSON schema strings as scalars.